autogpt/chat.py

```diff
@@ -140,7 +140,13 @@ def chat_with_ai(
                     "assistant", assistant_reply))
 
             return assistant_reply
-        except openai.error.RateLimitError:
+        except (openai.error.RateLimitError, openai.error.APIError) as exc:
```

Comment:
This looks my code edit, but I get zero attribution with this new PR you've proposed.
Who cares anyway.. the main importance is that it get's merged, but it's a shame that you dont give attribution to people's work @ryanpeach or maybe messed up some git commands, who knows. Anyways, I don't think creating yet another PR for the same thing makes sense getting it merged. Or is this PR doing something new? Can't get it out of the supreme PR decription ;)

Reply:
My point wasn't to cut you out. How would I add attribution other than the fact I have mentioned it several times in the PR?

Reply:
@ryanpeach

If PRs are not atomic, focused on a single thing, they put me in a difficult position when it comes to considering merging them.

I've been urged many times by members of the community that merging PRs that do this set a bad precedent and crate a broken git history. Additionally they make it difficult to remove problem causing PRs at a later date if necessary.

Atomic PRs that don't sneak in any extra little changes, that are well documented and only have diffs on the intended lines tend to get merged as soon as they've been tested - so long as they align with the project's direction.

Reply:
Ultimately this is my fault for not making the guidelines for contributors clear. 
I'll work on this.